src/geolocation.lua

```diff
@@ -0,0 +1,61 @@
+--[[
+@file geolocation.lua
+@license The MIT License (MIT)
+@copyright 2017 transitd
+--]]
+
+--- @module geolocation
+
+local geolocation = {}
+
+local config = require("config")
+local shell = require("lib.shell")
+local threadman = require("threadman")
+local support = require("support")
+
+function geolocation.run()
+
+    local listener = threadman.registerListener("geolocation",{"exit", "heartbeat"})
+
+    while true do
+        local msg = listener:listen()
+        if msg ~= nil then
+            if msg["type"] == "exit" then
+                break
+            end
+        end
+    end
+
+    if cmd then
+
+        local result = shrunner.execute(cmd)
```

Comment:
This code would not do anything useful because it would only run during exit from the above while loop which happens during program exit.